gui/wxpython/startup/utils.py

```diff
@@ -87,3 +87,34 @@ def create_database_directory():
         pass
 
     return None
+
+
+def startup_location_exists():
+    """Check for demolocation directory in distribution.
+
+    Returns True if found or False if nothing was found.
+    """
+    home = os.path.expanduser("~")
+    demolocation = os.path.join(home, "grass", "demolocation"),
```

Comment:
```
    home = os.path.expanduser("~")
    demolocation = os.path.join(home, "grass", "demolocation"),
```
Do I understand correctly that this is only for testing purposes ? Otherwise, a normal user will not have a $HOME/grass/demolocation directory.

More generally, I think we have to check on how to address this. As discussed during our last meeting, some distributions (e.g. Debian & derivatives) do not provide this demolocation when the user installs GRASS GIS. I sent a [mail on the grass-dev mailing list](https://lists.osgeo.org/pipermail/grass-dev/2020-August/094522.html) to get some feedback from packagers on the question on how to deal with data to be installed.

Reply:
> 
> 
> > I have a problem with testing that. Not able to get into the initial state.
> > While starting GRASS it displays _"No GRASS location found in .. Create new location or choose different GRASS GIS directory."_
> > The reason is that the processing in the function SuggestDatabase ends up on the row 513 and does not dive deeper to the function body. Therefore, it does not get a possible database path or does not create new grass database.
> 
> For me this was not the reason, but an incorrect usage of copytree(). See my comments in the code. With the suggested change, it seems to work for me.

Yes. I know, but I cannot check the copying since I cannot get to that state when copying should be performed. I have changed it according to your suggestion but my process still ends up on row 513 in the function SuggestDatabase so I cannot test it...

Reply:
> 
> 
> ```
>     home = os.path.expanduser("~")
>     demolocation = os.path.join(home, "grass", "demolocation"),
> ```
> 
> Do I understand correctly that this is only for testing purposes ? Otherwise, a normal user will not have a $HOME/grass/demolocation directory.
> 
> More generally, I think we have to check on how to address this. As discussed during our last meeting, some distributions (e.g. Debian & derivatives) do not provide this demolocation when the user installs GRASS GIS. I sent a [mail on the grass-dev mailing list](https://lists.osgeo.org/pipermail/grass-dev/2020-August/094522.html) to get some feedback from packagers on the question on how to deal with data to be installed.

I am not sure where in user's directory the demolocation will eventually exist. You have mentioned the directory usr/lib/grassXY/demolocation but I do not have any "grassXY" directory. I have only "grass". I looked at environmental variables, there is a variable GISBASE in my case /home/linwe/grass/dist.x86_64-pc-linux-gnu. Is it the variable I should use?

Reply:
Yes, it would be GISBASE. In grassXY X represents the major and Y the minor version as in many distributions GRASS GIS is installed at e.g. /usr/lib/grass78.

But getting the GISBASE content should be enough, I think.

Reply:
I think it is not the right one. I found on my computer in "computer/usr/lib/grass78" directory. and GISBASE is /home/linwe/grass/dist.x86_64-pc-linux-gnu for me. And also it was added into Debian installation but not Windows one. Am I right?

Reply:
Using `GISBASE` is the right thing to do now. Let's not worry about distributions/installations right now.